RocketChat/Rocket.Chat.ReactNative
# Chore: evaluate ServerItem - TypeScript Migration

## Proposed changes

Updates interfaces and removes `withTheme` HOC
## Issue(s)	


## How to test or reproduce

1. Press or long press a server item on the server dropdown
2. Change themes and check if server item on the server dropdown updates properly
## Screenshots

## Types of changes




## Checklist



## Further comments
